onecodex/distance.py

```diff
@@ -58,23 +51,28 @@ def beta_diversity(self, metric="braycurtis", rank="auto"):
         """
         import skbio.diversity
 
-        if metric not in ("jaccard", "braycurtis", "cityblock"):
+        if not BetaDiversityMetric.has_value(metric):
             raise OneCodexException(
-                "For beta diversity, metric must be one of: jaccard, braycurtis, cityblock"
+                "For beta diversity, metric must be one of: {}".format(
+                    ", ".join(BetaDiversityMetric.values())
+                )
             )
 
         df = self.to_df(rank=rank, normalize=self._guess_normalized())
 
-        counts = []
-        for c_id in df.index:
-            counts.append(df.loc[c_id].tolist())
+        if metric == BetaDiversityMetric.WeightedUnifrac:
+            return self.unifrac(weighted=True, rank=rank)
+        elif metric == BetaDiversityMetric.UnweightedUnifrac:
+            return self.unifrac(weighted=False, rank=rank)
+        elif metric == BetaDiversityMetric.Jaccard:
+            df = df > 0  # Jaccard requires a boolean matrix, otherwise it throws a warning
```

Comment:
`df > 0`... what does it even do? replace by a boolean every value that is > 0 in the dataframe?

Reply:
Yep, exactly. `jaccard` throws a warning if you pass it a quantitative data frame, since it then does the conversion implicitly.